From 6b4e2191481c4baaac369fb9496eab9b21921e05 Mon Sep 17 00:00:00 2001
From: redraz <redrazlive@gmail.com>
Date: Tue, 20 Aug 2024 13:39:03 +0300
Subject: [PATCH] Remove OLED restrictions + min 45Hz

---
 Makefile        |  2 +-
 source/main.cpp | 16 ++++++++--------
 2 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/Makefile b/Makefile
index a8d7720..1532333 100644
--- a/Makefile
+++ b/Makefile
@@ -38,7 +38,7 @@ include $(DEVKITPRO)/libnx/switch_rules
 #   NACP building is skipped as well.
 #---------------------------------------------------------------------------------
 APP_TITLE	:=	FPSLocker
-APP_VERSION	:=	2.0.1
+APP_VERSION	:=	2.0.1U
 
 TARGET		:=	FPSLocker
 BUILD		:=	build
diff --git a/source/main.cpp b/source/main.cpp
index d6539e5..f3e4049 100644
--- a/source/main.cpp
+++ b/source/main.cpp
@@ -770,7 +770,7 @@ public:
 			auto *clickableListItem = new tsl::elm::ListItem("Increase Refresh Rate");
 			clickableListItem->setClickListener([this](u64 keys) { 
 				if ((keys & HidNpadButton_A) && !isDocked) {
-					if ((this -> refreshRate >= 40) && (this -> refreshRate < 60)) {
+					if ((this -> refreshRate >= 45) && (this -> refreshRate < 60)) {
 						this -> refreshRate += 5;
 						if (!isDocked && R_SUCCEEDED(SaltySD_Connect())) {
 							SaltySD_SetDisplayRefreshRate(this -> refreshRate);
@@ -792,7 +792,7 @@ public:
 			auto *clickableListItem2 = new tsl::elm::ListItem("Decrease Refresh Rate");
 			clickableListItem2->setClickListener([this](u64 keys) { 
 				if ((keys & HidNpadButton_A) && !isDocked) {
-					if (this -> refreshRate > 40) {
+					if (this -> refreshRate > 45) {
 						this -> refreshRate -= 5;
 						if (!isDocked && R_SUCCEEDED(SaltySD_Connect())) {
 							SaltySD_SetDisplayRefreshRate(this -> refreshRate);
@@ -823,7 +823,7 @@ public:
 						u64 PID = 0;
 						Result rc = pmdmntGetApplicationProcessId(&PID);
 						if (!isDocked && R_SUCCEEDED(rc) && FPSlocked_shared) {
-							if (!displaySync == true && *FPSlocked_shared < 40) {
+							if (!displaySync == true && *FPSlocked_shared < 45) {
 								SaltySD_SetDisplayRefreshRate(60);
 								*displaySync_shared = 0;
 								refreshRate_g = 0;
@@ -869,7 +869,7 @@ public:
 			isDocked = false;
 		}
 		if (!isDocked)
-			snprintf(refreshRate_c, sizeof(refreshRate_c), "LCD Refresh Rate: %d Hz", refreshRate);
+			snprintf(refreshRate_c, sizeof(refreshRate_c), "Screen Refresh Rate: %d Hz", refreshRate);
 		else strncpy(refreshRate_c, "Not available in docked mode!", 30);
 	}
 };
@@ -1067,7 +1067,7 @@ public:
 					}
 					if (!isDocked && !oldSalty && displaySync) {
 						if (R_SUCCEEDED(SaltySD_Connect())) {
-							if (*FPSlocked_shared >= 40) {
+							if (*FPSlocked_shared >= 45) {
 								SaltySD_SetDisplayRefreshRate(*FPSlocked_shared);
 								refreshRate_g = *FPSlocked_shared;
 							}
@@ -1100,7 +1100,7 @@ public:
 					}
 					if (!isDocked && !oldSalty && displaySync) {
 						if (R_SUCCEEDED(SaltySD_Connect())) {
-							if (*FPSlocked_shared >= 40) {
+							if (*FPSlocked_shared >= 45) {
 								SaltySD_SetDisplayRefreshRate(*FPSlocked_shared);
 								refreshRate_g = *FPSlocked_shared;
 							}
@@ -1255,12 +1255,12 @@ public:
 			apmInitialize();
 			setsysInitialize();
 			SetSysProductModel model;
-			if (R_SUCCEEDED(setsysGetProductModel(&model))) {
+			/*if (R_SUCCEEDED(setsysGetProductModel(&model))) {
 				if (model == SetSysProductModel_Aula) {
 					isOLED = true;
 					remove("sdmc:/SaltySD/flags/displaysync.flag");
 				}
-			}
+			}*/
 			setsysExit();
 			fsdevMountSdmc();
 			FILE* file = fopen("sdmc:/SaltySD/flags/displaysync.flag", "rb");
-- 
2.42.0

